### DISCUSSION TITLE
Initial Raspbian OS (Bullseye) testing

### DISCUSSION THREAD

**Discussion by solskogen** [BlitterStudio/amiberry#851] [General]
Environment: SDL 2.0.14 - Kernel 5.10.63-v8+

fkms - sdl2 - console = GUI shows, black screen during emulation
fkms - sdl2 - wayland (sway) = Works
fkms - sdl2 - x11 = Works

fkms - dmx - console = Mouse not responding in GUI (keyboard works) - emulation works
fkms - dmx - wayland (sway) = GUI shows, but keyboard nor mouse works (focus / grab problem?)
fkms - dmx - x11 = Works

kms - sdl2 - console = Works, but slow (30fps)
kms - sdl2 - wayland (sway) = Works
kms - sdl2 - x11 = Works (50fps)

**Comment by midwan** [comment]
> fkms - sdl2 - console = GUI shows, black screen during emulation

Check if you have the sdl2 thread option enabled in `amiberry.conf` - this usually doesn't work with KMSDRM, and causes a black screen in emulation as described. Disabling it should make it work, if that's the issue.

> fkms - dmx - console = Mouse not responding in GUI (keyboard works) - emulation works
> fkms - dmx - wayland (sway) = GUI shows, but keyboard nor mouse works (focus / grab problem?)

Not sure about these, the mechanism is identical in all environments: we just use SDL2 to capture any triggered events and handle them. If SDL2 doesn't trigger those events for some reason, Amiberry won't handle them. It seems to me this is more SDL2 related? The Dispmanx part is _only_ used to draw the pixels on screen, events are all still handled by SDL2...

> kms - sdl2 - console = Works, but slow (30fps)

Sounds like a performance problem with KMSDRM in this case. Nothing I can do about it I'm afraid...

**Reply by solskogen** [reply]
`pi@bullseye:~/amiberry-dev $ grep sdl2 conf/amiberry.conf
use_sdl2_render_thread=no`

**Comment by solskogen** [comment]
Everything works is expected if you use a newer SDL2 (2.0.18 and newer)
